crates/ruff/src/rules/flake8_bandit/rules/hardcoded_password_default.rs

```diff
@@ -39,29 +40,20 @@ fn check_password_kwarg(arg: &Arg, default: &Expr) -> Option<Diagnostic> {
 pub(crate) fn hardcoded_password_default(arguments: &Arguments) -> Vec<Diagnostic> {
     let mut diagnostics: Vec<Diagnostic> = Vec::new();
 
-    let defaults_start =
-        arguments.posonlyargs.len() + arguments.args.len() - arguments.defaults.len();
-    for (i, arg) in arguments
-        .posonlyargs
-        .iter()
-        .chain(&arguments.args)
-        .enumerate()
-    {
-        if let Some(i) = i.checked_sub(defaults_start) {
-            let default = &arguments.defaults[i];
-            if let Some(diagnostic) = check_password_kwarg(arg, default) {
-                diagnostics.push(diagnostic);
-            }
-        }
-    }
-
-    let defaults_start = arguments.kwonlyargs.len() - arguments.kw_defaults.len();
-    for (i, kwarg) in arguments.kwonlyargs.iter().enumerate() {
-        if let Some(i) = i.checked_sub(defaults_start) {
-            let default = &arguments.kw_defaults[i];
-            if let Some(diagnostic) = check_password_kwarg(kwarg, default) {
-                diagnostics.push(diagnostic);
-            }
+    for ArgWithDefault {
+        def,
+        default,
+        range: _,
+    } in chain!(
+        &arguments.posonlyargs,
+        &arguments.args,
+        &arguments.kwonlyargs
```

Comment:
What's the motivation for using `chain!` over `iter.chain`? I know we have a few use cases for `itertools` but I try to keep them to a minimum, in the hope, that we could someday remove the dependency.

Reply:
I can change it to `iter.chain`. I just wanted to be consistent everywhere and this was more concise.